### PR TITLE
RCHAIN- 3129 sig validation function

### PR DESCRIPTION
**Pull request by rabbitonweb**
## Overview

1.  Align messages in DeployData
2. Introduce SignatreuAlg as a common interface to (currently) Ed25519 a…
3. Reuse SignaturesAlg in SignatureAlgorithms  …

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3129

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/SignatureAlgorithms.scala [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
   type SignatureAlgorithm = (Array[Byte], Array[Byte]) => Array[Byte]
   @SuppressWarnings(Array("org.wartremover.warts.Throw")) // TODO remove throw
   def lookup(algorithm: String): SignatureAlgorithm =
-    algorithm match {
-      case "ed25519"   => Ed25519.sign
-      case "secp256k1" => Secp256k1.sign
-      case _           => throw new Exception(s"Unknown signature algorithm $algorithm")
-    }
+    SignaturesAlg(algorithm)
+      .fold(throw new Exception(s"Unknown signature algorithm $algorithm"))(_.sign)
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/util/SignatureAlgorithms.scala]: Are you sure this won't eagerly throw?

Comment (rabbitonweb) [reply casper/src/main/scala/coop/rchain/casper/util/SignatureAlgorithms.scala]: it will, is it a bad thing?

Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/util/SignatureAlgorithms.scala]: I'm not sure how fold is implemented but I thought it does something like calculate the default value and then pass that into the accumulator. `_.sign` then just ignores the value of the accumulator but it doesn't change the fact that the `throw new Exception` was executed. So I guess the exception isn't thrown unless it is returned?

Comment (odeac) [reply casper/src/main/scala/coop/rchain/casper/util/SignatureAlgorithms.scala]: scala> val opt = Some(10)
opt: Some[Int] = Some(10)
scala> opt.fold(throw new Exception("aaa")) (identity)
res3: Int = 10

**Pull request merged**